### PR TITLE
Parser: more cleanup

### PR DESCRIPTION
**Pull request by dnicolodi**
Some minor stuff that annoyed me during the port of the lexer from Flex to RE/flex.

### CODE REVIEW

**Comment by dnicolodi** [comment]
I am pretty sure the pylint error is not introduced by these patches.

File: beancount/parser/lexer.l [hunk 1]
```diff
@@ -240,34 +235,31 @@ NULL		{
 ^[\*\:\#]/.	{ BEGIN(IGNORE); }
 ^{FLAGS}/.	{ BEGIN(IGNORE); }
 
- /* Default rule. {bf253a29a820} */
```
Comment (blais) [inline beancount/parser/lexer.l:243]: Please leave the reference in (grep to find the other side).

**Comment by blais** [comment]
I'll add it back in myself.
In the future, please LMK AOT before updating the versions of the generators.
Probably best to do those on their own.

**Comment by dnicolodi** [comment]
> In the future, please LMK AOT before updating the versions of the generators.

Sure. It is not really needed in this case as the new version of Bison does not bring anything relevant to the use in beancount. I just upgraded my system and I was too lazy to put together an environment with the old version.

**Pull request merged**